### ISSUE TITLE
Investigate integrating support for Genie (Cisco's open source device output parser library)

### ISSUE THREAD

**Issue by jesse-mcgraw** [ktbyers/netmiko#1237]
Integrate support for this parser library in a similar fashion to TextFsm

More info on Genie and pyats: 
  https://developer.cisco.com/docs/pyats/

Food for thought on parsing command output via the parse_genie ansible module:
  https://github.com/clay584/parse_genie/blob/master/filter_plugins/genie.py

**Comment by carlniger** [comment]
@jesse-mcgraw maybe a dumb question since I've not actually done any hands on with genie at this point, but could you summarize the benefit/difference between genie and TextFSM?

**Comment by jesse-mcgraw** [comment]
Both have very similar overall goals: parse unstructured CLI data into structured data

The advantages I see are:
- Cisco is maintaining this library and constantly updating parsers
- output is in a more consistent and logical format, they're aligning it with YANG

disadvantages
- Only supports Cisco stuff out of the box.  Creating parsers does not seem very difficult but upstreaming them probably would be

**Comment by carlniger** [comment]
@jesse-mcgraw can you clone my commit for #1240 and test it out and see how it works? I tested on some ios-xe boxes and it seems to work as desired.

**Comment by carlniger** [comment]
Continuing conversation in the PR, closing!
